op-node/rollup/derive/types.go

```diff
@@ -0,0 +1,63 @@
+package derive
+
+import (
+	"fmt"
+	"regexp"
+)
+
+type CompressionAlgo string
+
+const (
+	// compression algo types
+	Zlib     CompressionAlgo = "zlib"
+	Brotli9  CompressionAlgo = "brotli-9"
+	Brotli10 CompressionAlgo = "brotli-10"
+	Brotli11 CompressionAlgo = "brotli-11"
+)
+
+var CompressionAlgoTypes = []CompressionAlgo{
+	Zlib,
+	Brotli9,
+	Brotli10,
+	Brotli11,
+}
+
+var brotliRegexp = regexp.MustCompile(`^brotli-(9|10|11)$`)
+
+func (algo CompressionAlgo) String() string {
+	return string(algo)
+}
+
+func (algo *CompressionAlgo) Set(value string) error {
+	if !ValidCompressionAlgoType(CompressionAlgo(value)) {
+		return fmt.Errorf("unknown compression algo type: %q", value)
+	}
+	*algo = CompressionAlgo(value)
+	return nil
+}
```

Comment:
is this used anywhere?

Reply:
This is used for `cli.Generic` in the cmd flag